src/app/core/ui/navigation/menu-item/menu-item.component.html

```diff
@@ -23,9 +23,21 @@
 </mat-list-item>
 
 <!-- Render submenus if they exist and are expanded -->
-<div *ngIf="isExpanded" class="submenu">
-  <ng-container *ngFor="let subItem of item.subMenu">
-    <!-- Pass activeLink to each submenu item to highlight the active submenu -->
-    <app-menu-item [item]="subItem" [activeLink]="activeLink"></app-menu-item>
+<div *ngIf="isExpanded" class="submenu" cdkDropList (cdkDropListDropped)="dropSubMenu($event)"  cdkDropListOrientation="vertical"
+cdkDropListLockAxis="y"
+class="drop-list flex-column gap-regular">
+  <ng-container *ngFor="let subItem of item.subMenu; let i = index">
+    <div class="submenu-item-container" cdkDrag>
+      <fa-icon icon="grip-vertical" class="drag-handle"></fa-icon>
+
+      <app-menu-item [item]="subItem" [activeLink]="activeLink"></app-menu-item>
+
+      <fa-icon
+        icon="times"
+        class="remove-icon"
+        (click)="removeSubMenuItem(i)"
+      ></fa-icon>
+    </div>
   </ng-container>
 </div>
+
```

Comment:
I think this is the cause for the actual menu looking weird. Can you revert the changes here?
We'll implement a fresh solution for administering nested items then.